server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/LinshareBlobExportMechanismIntegrationTest.java

```diff
@@ -0,0 +1,274 @@
+    @BeforeEach
+    void setup(GuiceJamesServer jmapServer) throws Throwable {
+        this.jmapServer = jmapServer;
+
+        jmapServer.getProbe(DataProbeImpl.class)
+            .fluent()
+            .addDomain(DOMAIN)
+            .addUser(HOMER, HOMER_PASSWORD)
+            .addUser(BART, BART_PASSWORD);
+
+        RestAssured.requestSpecification = jmapRequestSpecBuilder
+            .setPort(jmapServer.getProbe(JmapGuiceProbe.class).getJmapPort())
+            .build();
+        RestAssured.defaultParser = Parser.JSON;
+
+        webAdminApi = WebAdminUtils.spec(jmapServer.getProbe(WebAdminGuiceProbe.class).getWebAdminPort());
+
+        MailboxProbe mailboxProbe = jmapServer.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, HOMER, DefaultMailboxes.INBOX);
+
+        homerAccessToken = authenticateJamesUser(baseUri(jmapServer), HOMER, HOMER_PASSWORD);
+        bartAccessToken = authenticateJamesUser(baseUri(jmapServer), BART, BART_PASSWORD);
+        user1LinshareAPI = new LinshareExtension().getAPIFor(USER_1);
+
+        fakeSmtpRequestSpecification = given(new LinshareExtension().getLinshare().fakeSmtpRequestSpecification());
+    }
+
+    @Test
+    void exportShouldShareTheDocumentViaLinShareWhenJmapDelete() {
+        bartSendMessageToHomer();
+
+        WAIT_TEN_SECONDS.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+
+        homerDeletesMessages(listMessageIdsForAccount(homerAccessToken));
+        WAIT_TEN_SECONDS.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 0);
+
+        exportVaultContent(EXPORT_ALL_HOMER_MESSAGES_TO_USER_1);
+
+        assertThat(user1LinshareAPI.receivedShares())
+            .hasSize(1)
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getDocument().getName()).endsWith(".zip"))
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getSender().getMail()).isEqualTo(USER_1.getUsername()));
+    }
+
+    @Test
+    void exportShouldShareTheDocumentViaLinShareWhenImapDelete() throws Exception {
+        bartSendMessageToHomer();
+
+        WAIT_TEN_SECONDS.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 1);
+
+        imapMessageReader.connect(LOCALHOST_IP, jmapServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(HOMER, HOMER_PASSWORD)
+            .select(IMAPMessageReader.INBOX)
+            .setFlagsForAllMessagesInMailbox("\\Deleted");
+        imapMessageReader.expunge();
+
+        WAIT_TEN_SECONDS.until(() -> listMessageIdsForAccount(homerAccessToken).size() == 0);
+
+        exportVaultContent(EXPORT_ALL_HOMER_MESSAGES_TO_USER_1);
+
+        assertThat(user1LinshareAPI.receivedShares())
+            .hasSize(1)
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getDocument().getName()).endsWith(".zip"))
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getSender().getMail()).isEqualTo(USER_1.getUsername()));
```

Comment:
how do you ensure the zip is not empty?

Reply:
The thing is I need to add an extra request to download the zip from Linshare and verify it works on the different layers of tests. I propose you to make that in an other PR, so that we can divide the work and validate this part already

Reply:
=>  https://github.com/linagora/james-project/pull/2340